File: chapter 5/insertion.py
def updateBits(n, m, i, j):
    allOnes = ~0

    left = allOnes << (j+1)

    right = ((1 << i) -1)

    mask = left | right

    n_cleared = m & mask
    m_shifted = m << i

    return n_cleared | m_shifted

print updateBits(int('10000000000', 2), int('10011', 2), 2, 6)

File: chapter 5/insertion.cpp
// #include <bits/stdc++>;
#include <iostream>
using namespace std;

int updateBits(long long int n, int m, int i, int j){
    int allOnes = ~0;

    int left = allOnes << (j+1);

    int right = ((1 << i) - 1);

    int mask = left | right;

    int n_cleared = n & mask;
    int m_shifted = m << 1;
    return n_cleared | m_shifted;
}

int main(){
    cout << updateBits(10000000000, 10011, 2, 6) << endl;
    return 0;
}